### ISSUE TITLE
Data Ownership

### ISSUE THREAD

**Issue by nkallergis** [nautobot/nautobot-app-design-builder#55]
### Environment
* Nautobot version:  2.x
* design-builder version:  next

### Proposed Functionality
Building on issue #52.

For all the data managed by a service, defined in the journal attribute, we should avoid that other mechanisms can change this data outside of the design builder. This enforcement should happen in the related objects, at different levels (object or attribute or key in an attribute), so an option could be using Custom Validators which will take into account the references to designs. For instance, using a relationship towards the design at object level, it could check the related journals (using the reverse key to AssociatedDesignObject), to see if the data is protected or not. In more detail, the process could look like:

1. When an object is going to call the validation, it would check the Custom Validator
2. If there is a reference from this object to one or many designs, it would extract the Journals
3. It would check if the change requested is compatible with the Journals, and if not, it would reject the change. For instance, if it’s trying to change a key in a config context which is owned by a design, it should say that this key can’t be overwritten because a design owns this data.

### Use Case
As Bob the Design Builder, I'd like to have the capability to "lock" data that are solely managed by my designs and prevent user manipulation. For instance:
```
{
  “description”: "this is an important device”,
 “config_context”: {“lldp”: true}
}
```
For the device config context, because we are only referencing the “lldp” key, this is the only key that is owned by the design. Other keys in the config_context can be modified outside of the control of this design.

**Comment by chadell** [comment]
this is already in place in `feature_delices` branch.

**Comment by abates** [comment]
data protection has been implemented in the design lifecycle branch. Already available in ltm-1.6 and soon in 2.x
